daemon/container.go

```diff
@@ -1137,6 +1138,31 @@ func (container *Container) shouldRestart() bool {
 		(container.hostConfig.RestartPolicy.Name == "on-failure" && container.ExitCode != 0)
 }
 
+func (container *Container) mountVolumes() error {
```

Comment:
```go
(*Container).ArchivePath(path string) (content io.ReadCloser, stat *types.ContainerPathStat, err error)
```
This method avoids that issue by naming the returned error. That way the `err` check in the `defer` will be equal the returned `err`.